patch/kernel/archive/bcm2711-5.19/1003-remote-wakeup.patch

```diff
@@ -0,0 +1,160 @@
+diff --git a/drivers/usb/dwc2/gadget.c b/drivers/usb/dwc2/gadget.c
+index 0909b088a284..31ff5348020c 100644
+--- a/drivers/usb/dwc2/gadget.c
++++ b/drivers/usb/dwc2/gadget.c
+@@ -4763,6 +4763,78 @@ static void dwc2_gadget_set_speed(struct usb_gadget *g, enum usb_device_speed sp
+ 	spin_unlock_irqrestore(&hsotg->lock, flags);
+ }
+ 
++/**
++ * dwc2_hsotg_wakeup - send wakeup signal to the host
++ * @gadget: The usb gadget state
++ *
++ * If the gadget is in device mode and in the L1 or L2 state,
++ * it sends a wakeup signal to the host.
```

Comment:
This is the task of interaction between the host software and the device software.
Or if the host is the operating system on the device.
Example: I hold the phone in my hands and he usually sleeps. If I connect it via usb,
it will become a 4G router and wake up.